### PR TITLE
add function for conditional validation

### PR DESCRIPTION
**Pull request by chlebowa**
Introducing `crule` function to modify rule funcitons for input validation by `shinyvalidate`.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -33,3 +35,65 @@ get_teal_bs_theme <- function() {
...
+  checkmate::assert_multi_class(rule, c("function", "formula"))
+  checkmate::assert_class(substitute(condition), "call")
+
+  if (inherits(rule, "formula")) {
+    rule <- rlang::as_function(rule)
+  }
+
+  dep <- paste(deparse(substitute(condition)), collapse = "")
+  ex <- list(
+    str2lang(sprintf("if (isFALSE(%s)) return(NULL)", dep)),
```
Comment (gogonzo) [inline R/utils.R:94]: you can work with language directly like this:
```r
cond <- quote(a == "b")
substitute(if (xxx) "true" else "false", list(xxx = cond))
# if (a == "b") "true" else "false"

```

```suggestion
  cond <- substitute(condition) # language
  ex <- list(
    substitute(if (isFALSE(cond)) return(NULL), list(cond = cond)),
```

File: R/utils.R [hunk 2]
```diff
@@ -33,3 +35,65 @@ get_teal_bs_theme <- function() {
...
+    body(rule)
+  )
+  body(rule) <- as.call(c(as.name("{"), ex))
+  rule
```
Comment (gogonzo) [inline R/utils.R:98]: You don't have to convert to character - use substitute instead

```suggestion
  body(rule) <- call(
    "{",
    substitute(if (isFALSE(condition)) return(NULL)),
    body(rule)
  )
  rule
```

Comment (chlebowa) [reply R/utils.R:98]: Thanks, I tried and failed to do just that 👍

**Comment by chlebowa** [comment]
After deliberating with @gogonzo we decided that this is syntactic sugar and that we can do without it.

**Pull request closed without merging**